plugins/modules/keycloak_clientsecret.py

```diff
@@ -0,0 +1,215 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2022, Fynn Chen <ethan.cfchen@gmail.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+DOCUMENTATION = '''
+---
+module: keycloak_clientsecret
+
+short_description: Administration of Keycloak client secret via Keycloak API
+
+version_added: 6.1.0
+
+description:
+  - This module allows you to get or generate new Keycloak client secret via the Keycloak REST API.
+    It requires access to the REST API via OpenID Connect; the user connecting and the client being
+    used must have the requisite access rights. In a default Keycloak installation, admin-cli
+    and an admin user would work, as would a separate client definition with the scope tailored
+    to your needs and a user having the expected roles.
+
+  - The names of module options are snake_cased versions of the camelCase ones found in the
+    Keycloak API and its documentation at U(https://www.keycloak.org/docs-api/8.0/rest-api/index.html).
+
+  - Attributes are multi-valued in the Keycloak API. All attributes are lists of individual values and will
+    be returned that way by this module. You may pass single values for attributes when calling the module,
+    and this will be translated into a list suitable for the API.
+
+  - When generate a new client secret, where possible provide the client ID (not client_id) to the module.
+    This removes a lookup to the API to translate the client_id into the client ID.
+
+
+options:
+  state:
+    description:
+      - State of the client secret.
+      - On C(present), get the current client secret.
+      - On C(absent), the new client secret will be generated.
```

Comment:
Hmm, the whole interface is a bit strange. `state=present` is a get operation, and `state=absent` is a regenerate operation? That sounds weird to me.

I wonder whether there should instead be two modules, `keycloak_clientsecret_regenerate` which regenerates the secret, and `keycloak_clientsecret_info` which retrieves the current secret.

Reply:
Agree the behavior of `absent` doesn't make sense. Would you be happy with me adding a `regenerate` boolean parameter and only allowing `state=present` instead of splitting across two modules?

Reply:
`state=present` should always be "create something if needed". This module is about a resource which appaently cannot be created, only re-created. So `state=present` should actually be called `state=get`, but `state=get` is not allowed and should be moved to an `_info` module (https://docs.ansible.com/ansible/devel/dev_guide/developing_modules_best_practices.html#scoping-your-module-s).

Reply:
Ok - in that case I'll split into those two separate modules.